src/main/resources/hudson/plugins/git/GitPublisher/config.jelly

```diff
@@ -70,6 +70,11 @@
                     add="${%Add Branch}">
         <table width="100%">
           <br/>
+          <f:entry field="srcBranchName"
+                   title="${%Branch for push}">
```

Comment:
Agreed.
Probably, you also need an additional validation handler. Source branch must always exist, but the destination branch can be created on-demand